File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/StatisticDao.java
package com.platdmit.simplecloudmanager.data.database.dao;

import com.platdmit.simplecloudmanager.data.database.entity.DbStatistic;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Query;

@Dao
public interface StatisticDao extends BaseDao<DbStatistic>{
    @Query("SELECT * FROM dbstatistic WHERE mServer = :serverId")
    List<DbStatistic> getStatisticsForServer(long serverId);

    @Query("DELETE FROM dbstatistic WHERE mServer = :serverId")
    void deleteAll(long serverId);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/StatisticDao.kt
package com.platdmit.simplecloudmanager.data.database.dao

import androidx.room.Dao
import androidx.room.Query
import com.platdmit.simplecloudmanager.data.database.entity.DbStatistic

@Dao
interface StatisticDao : BaseDao<DbStatistic> {
    @Query("SELECT * FROM dbstatistic WHERE server = :serverId")
    fun getStatisticsForServer(serverId: Long): List<DbStatistic>?

    @Query("DELETE FROM dbstatistic WHERE server = :serverId")
    fun deleteAll(serverId: Long)
}